src/cast_core/src/actors/CrudActor.cpp

```diff
@@ -0,0 +1,1077 @@
+            throw InvalidConfigurationException(
+                "'insertMany' expects a 'Documents' field of sequence type.");
+        }
+        for (auto&& document : documents) {
+            _docTemplates.push_back(std::move(value_generators::makeDoc(document, rng)));
+        }
+        // TODO: parse insert options.
+    }
+
+    void run(mongocxx::client_session& session) override {
+        for (auto&& docTemplate : _docTemplates) {
+            bsoncxx::builder::stream::document doc{};
+            docTemplate->view(doc);
+            _writeOps.push_back(doc.extract());
+        }
+        auto ctx = _operation.start();
+        (_onSession) ? _collection.insert_many(session, _writeOps, _options)
+                     : _collection.insert_many(_writeOps, _options);
+        ctx.success();
+    }
+
+private:
+    mongocxx::collection _collection;
+    const bool _onSession;
+    std::vector<bsoncxx::document::value> _writeOps;
+    mongocxx::options::insert _options;
+    metrics::Operation _operation;
+    std::vector<UDocGenerator> _docTemplates;
+};
+
+/**
+ * Example usage:
+ *    Operations:
+ *    - OperationName: startTransaction
+ *      OperationCommand:
+ *        Options:
+ *          WriteConcern:
+ *            Level: majority
+ *            Journal: true
+ *          ReadConcern:
+ *            Level: snapshot
+ *          ReadPreference:
+ *            ReadMode: primaryPreferred
+ *            MaxStalenessSeconds: 1000
+ */
+
+struct StartTransactionOperation : public BaseOperation {
+
+    StartTransactionOperation(YAML::Node opNode,
+                              bool onSession,
+                              mongocxx::collection collection,
+                              genny::DefaultRandom& rng,
+                              metrics::Operation operation) {
+        if (!opNode.IsMap())
+            return;
+        if (opNode["Options"]) {
+            _options = opNode["Options"].as<mongocxx::options::transaction>();
+        }
+    }
+
+    void run(mongocxx::client_session& session) override {
+        session.start_transaction(_options);
+    }
+
+private:
+    mongocxx::options::transaction _options;
+};
+
+/**
+ * Example usage:
+ *    Operations:
+ *    - OperationName: commitTransaction
+ */
+
+struct CommitTransactionOperation : public BaseOperation {
+
+    CommitTransactionOperation(YAML::Node opNode,
+                               bool onSession,
+                               mongocxx::collection collection,
+                               genny::DefaultRandom& rng,
+                               metrics::Operation operation) {}
+
+    void run(mongocxx::client_session& session) override {
+        session.commit_transaction();
+    }
+};
+
+/**
+ * Example usage:
+ *    Operations:
+ *    - OperationName: setReadConcern
+ *      OperationCommand:
+ *        ReadConcern:
+ *          Level: majority
+ */
+
+struct SetReadConcernOperation : public BaseOperation {
```

Comment:
[question] Why did this end up being a separate operation instead of additional options to the commands? I was expected there'd be a way to make use of `ConfigNode` so the read and write concerns wouldn't need to be repeated for each operation in the list.

Reply:
My rationale for adding a separate operation was mostly to reflect the mongocxx::collections api command as it seemed like something that might be useful. I will add a note in the TODO ticket to consider adding the ability to set ReadConcern at the Phase level.